numpyro/svi.py

```diff
@@ -52,37 +60,52 @@ def init_fn(rng, model_args=(), guide_args=(), params=None):
             guide_init = substitute(guide_init, params)
         guide_trace = trace(guide_init).get_trace(*guide_args, **kwargs)
         model_trace = trace(model_init).get_trace(*model_args, **kwargs)
+        inv_transforms = {}
         for site in list(guide_trace.values()) + list(model_trace.values()):
             if site['type'] == 'param':
-                params[site['name']] = site['value']
-        return optim_init(params)
+                constraint = site['kwargs'].pop('constraint', constraints.real)
+                transform = biject_to(constraint)
+                inv_transforms[site['name']] = transform
+                params[site['name']] = transform.inv(site['value'])
+
+        def transform_constrained(inv_transforms, params):
+            return {k: inv_transforms[k](v) for k, v in params.items()}
```

Comment:
can we reuse `constrain_fn` logic in `hmc_util`?

Reply:
I have moved this to `infer_util` now, and we can put more common utilities there for prediction, etc in the future.